### PR TITLE
Datumize antags

### PR DESCRIPTION
**Pull request by SteelSlayer**
**What does this PR do:**
Include a small to medium description of what your PR changes. Feel free to give it a little flourish if you want to show off a new feature more completely.

**Images of sprite/map changes (IF APPLICABLE):**
If you did not make a map or sprite edit, you may leave this section blank. You may include a gif of your feature if you want

**Changelog:**
*Remove this line, and appropriate fields from the changelog*
:cl:
add: Added new things
del: Removed old things
tweak: tweaked a few things
balance: rebalanced something
fix: fixed a few things
wip: added a few works in progress
soundadd: added a new sound thingy
sounddel: removed an old sound thingy
imageadd: added some icons and images
imagedel: deleted some icons and images
spellcheck: fixed a few typos
experiment: added an experimental thingy
/:cl:

### CODE REVIEW

File: code/modules/admin/topic.dm [hunk 1]
```diff
@@ -1959,22 +1959,19 @@
...
-					kill_objective.explanation_text = "Assassinate [H.real_name], the [H.mind.assigned_role]."
-					newtraitormind.objectives += kill_objective
-					SSticker.mode.equip_traitor(newtraitor)
-					SSticker.mode.traitors |= newtraitor.mind
+					var/datum/antagonist/traitor/T = new()			
+					T.give_objectives = FALSE
+					T.should_give_codewords = FALSE			
+					T.should_equip = FALSE
+					// Stuck calling the proc below, doesnt matter if it goes after add_antag_datum(T) or not
+					T.forge_single_custom_objective(H.mind, "Assassinate [H.real_name], the [H.mind.assigned_role].")
```
Comment (SteelSlayer) [inline code/modules/admin/topic.dm]: Game gets stuck on this function

File: code/modules/antagonists/traitor/traitor.dm [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+/datum/antagonist/traitor/proc/forge_single_objective()
+	switch(traitor_kind)
+		if(TRAITOR_AI)
+			return forge_single_AI_objective()
+		else
+			return forge_single_human_objective()
+
+
+// Used for situations when you want to give someone only one objective. Apprentices, mindslaves, missionary staff slaves, crew traitors, syndicate beacon traitors.
+/datum/antagonist/traitor/proc/forge_single_custom_objective(var/objective_target, var/explanation)
```
Comment (SteelSlayer) [inline code/modules/antagonists/traitor/traitor.dm]: This never executes

File: code/datums/mind.dm [hunk 3]
```diff
@@ -800,7 +800,8 @@
 				qdel(flash)
 				take_uplink()
 				var/fail = 0
-				fail |= !current.mind.add_antag_datum(/datum/antagonist/traitor)
+				var/datum/antagonist/traitor/T = current
```
Comment (Couls) [inline code/datums/mind.dm:815]: don't think you should be passing current as an argument here, either just use new() or use has_antag_datum(/datum/antagonist/traitor) as that returns the minds traitor datum

File: code/datums/mind.dm [hunk 4]
```diff
@@ -1611,36 +1606,25 @@
...
-
-	missionary.mind.add_antag_datum(/datum/antagonist/traitor/custom)
-
-	if(missionary.mind.som)//do not add if not a traitor..and you just picked up a robe and staff in the hall...
+	
+	var/datum/antagonist/traitor/T = new()
+	T.give_objectives = FALSE
+	T.should_equip = FALSE
+	T.should_give_codewords = FALSE
+	T.forge_single_custom_objective(src, "Obey every order from and protect [missionary.real_name], the [missionary.mind.assigned_role == missionary.mind.special_role ? (missionary.mind.special_role) : (missionary.mind.assigned_role)].")
```
Comment (Couls) [inline code/datums/mind.dm]: pass /datum/objective/custom_objective to this instead of this spaghet

File: code/game/machinery/syndicatebeacon.dm [hunk 5]
```diff
@@ -73,18 +72,15 @@
 				else
 					objective = "Make certain at least 80% of the station evacuates on the shuttle."
 			
-			// var/datum/objective/escape/escape_objective = new
-			// N.add_objective(escape_objective)
+			var/datum/antagonist/traitor/T = new()
+			T.give_objectives = FALSE
+			T.forge_single_custom_objective(null, objective)
```
Comment (Couls) [inline code/game/machinery/syndicatebeacon.dm]: nice explanation text here (????)

File: code/modules/admin/topic.dm [hunk 6]
```diff
@@ -1959,22 +1959,19 @@
...
+					newtraitor.mind.add_antag_datum(T)
 					to_chat(newtraitor, "<span class='danger'>ATTENTION:</span> It is time to pay your debt to the Syndicate...")
-					to_chat(newtraitor, "<B>You are now a traitor.</B>")
-					to_chat(newtraitor, "<B>Goal: <span class='danger'>KILL [H.real_name]</span>, currently in [get_area(H.loc)]</B>");
-					newtraitor.mind.special_role = SPECIAL_ROLE_TRAITOR
-					var/datum/atom_hud/antag/tatorhud = huds[ANTAG_HUD_TRAITOR]
-					tatorhud.join_hud(newtraitor)
-					set_antag_hud(newtraitor, "hudsyndicate")
+					to_chat(newtraitor, "<B>Goal: <span class='danger'>KILL [H.real_name]</span>, currently in [get_area(H.loc)]</B>")
```
Comment (Couls) [inline code/modules/admin/topic.dm]: why not just announce_objectives() ?

File: code/modules/antagonists/traitor/traitor.dm [hunk 7]
```diff
@@ -13,17 +13,8 @@
 	var/should_equip = TRUE
 	var/traitor_kind = TRAITOR_HUMAN //Set on initial assignment
 	var/list/assigned_targets = list()
-	//can_hijack = HIJACK_HIJACKER
 
 
-// This is for dealing with antags like wizard apprentices and missionaries
-// We dont want to give these traitors codewords, uplinks or the normal assasinate or theft objectives
-/datum/antagonist/traitor/custom
```
Comment (Couls) [inline code/modules/antagonists/traitor/traitor.dm]: Could've kept this and given it to mindslaves, and missionaries

**Review by Couls — CHANGES_REQUESTED** [review]

**Pull request closed without merging**